### ISSUE TITLE
NVDA doesn't read Span text consistently when moving from focus into focusable span

### ISSUE THREAD

**Issue by LSchroeder-Epic** [nvaccess/nvda#14879]
### Steps to reproduce:

Load HTML structured like:

```
<div style="text-align:left;" tabindex="0">
	<div tabindex="0">
		<span class="icon" aria-label="Special" role="img"></span> 
		<span>Row 1 (20) 
			<span class="icon" tabindex="0" aria-label="New results" role="img"></span>
		</span>
	</div>
	<div tabindex="0">
		<span class="icon"></span>
		<span>Other (3)</span>
	</div>
</div>
```

Then using tab/shift-tab to move focus through the structure

### Actual behavior:
Inconsistent reading experience.

Sometimes will read both "lines" when focusing the div, sometimes only one of the other.
More often than not when tabbing into child div containing a line of data the first line will fail to read. 

Speech Viewer output
document
<Tab>
Special  graphic    Row 1 (20)  New results  graphic    Other (3)
<Tab>
<Tab>
New results  graphic (sometimes)
<Tab>
Other (3)

### Expected behavior:
Consistent behavior reading a div based structure containing multiple lines of data.

Speech Viewer output
document
<Tab>
Special  graphic    Row 1 (20)  New results  graphic    Other (3)
<Tab>
graphic    Special  Row 1 (20)  graphic    New results
<Tab>
New results  graphic
<Tab>
Other (3)

#### Name and version of other software in use when reproducing the issue: 
Microsoft Edge 112.0.1722.58 (64-bit)
Google Chrome 112.0.5615.138 (32-bit)

**Comment by Adriani90** [comment]
This happens in Firefox too.

It seems NVDA reports all nested divs / spans when entering them with tab or shift+tab. I think this is intended behavior.
@jcsteh any idea why NVDA e.g. reports all of them when entering the first parent div?

**Comment by jcsteh** [comment]
In browse mode, NVDA reports the content of a container when it gets focus. If your div or span contains a lot of other divs or spans, all of them will be reported, since they are the content of the focusable container.

**Comment by Adriani90** [comment]
Hmm ok thanks for the explanation. In Chrome Canary 133 it seems divs are focusable in focus mode and the whole container is also reported when using focus mode. This might be a rendering difference to Firefox.
Quite verbose though especially when web authors use containers with lots of nested focusable divs.

**Comment by jcsteh** [comment]
In focus mode, this is different in Chrome because Chrome computes the name of focusable divs/spans using the entire subtree, whereas Firefox doesn't. Strictly speaking, Chrome's behaviour is incorrect according to the spec, but there are multiple open questions being discussed in this area of the spec at the moment, so this may change. Either way, the accessibility implications of focusing a div or a span are not well defined, so this is best avoided unless you have a very, very good reason to do it.
